### PR TITLE
Eliminate compiler warnings

### PR DESCRIPTION
**Pull request by vacantron**
The cfront does not support casting yet. Skip it using the preprocessing.

### CODE REVIEW

File: src/defs.h [hunk 1]
```diff
@@ -45,6 +45,16 @@
...
 #define REG_CNT 8
 
+/* This macro will be automatically defined at shecc run-time. */
+#ifdef __SHECC__
+#define UNUSED(x) (x)
+#define SHECC_CAST(x) (x)
+#else
+/* suppress GCC/Clang warnings */
+#define UNUSED(x) (void) (x)
+#define SHECC_CAST(x) (void *) (x)
```
Comment (jserv) [inline src/defs.h:55]: I don't get the idea why `SHECC_CAST` exists. Can you explain?

Comment (jserv) [reply src/defs.h:55]: Is it necessary?

Comment (vacantron) [reply src/defs.h:55]: > I don't get the idea why `SHECC_CAST` exists. Can you explain?

In printf/fprintf, GCC/Clang will check the format string specified with `-Wall` flag
```c
fprintf(fd, "subgraph cluster_%p {\n", SHECC_CAST(bb));
```

> Is it necessary?

Or we could simply apply the `-Wno-format` flag.

Comment (jserv) [reply src/defs.h:55]: > Or we could simply apply the `-Wno-format` flag.

Let's enforce `-Wno-format` option.

**Pull request merged**